src/textual/css/stylesheet.py

```diff
@@ -433,6 +451,40 @@ def apply(
                 for name, specificity_rules in rule_attributes.items()
             },
         )
+
+        # Set initial values
+        for initial_rule_name in initial:
+            # Rules with a value of None should be set to the default value
+            if node_rules[initial_rule_name] is None:  # type: ignore[literal-required]
+                # Exclude non default values
+                # rule[0] is the specificity, rule[0][1] is 0 for default
+                default_rules = [
+                    rule
+                    for rule in rule_attributes[initial_rule_name]
+                    if not rule[0][0]
+                ]
+                if default_rules:
+                    # There is a default value
+                    new_value = max(default_rules, key=get_first_item)[1]
+                    node_rules[initial_rule_name] = new_value  # type: ignore[literal-required]
+                else:
+                    # No default value
+                    initial_defaults.add(initial_rule_name)
+
+        # Rules in DEFAULT_CSS set to initial
+        for initial_rule_name in initial_defaults:
+            if node_rules[initial_rule_name] is None:  # type: ignore[literal-required]
+                default_rules = [
+                    rule for rule in rule_attributes[initial_rule_name] if rule[0][0]
+                ]
+                if default_rules:
+                    # There is a default value
+                    new_value = max(default_rules, key=get_first_item)[1]
+                    node_rules[initial_rule_name] = new_value  # type: ignore[literal-required]
+                else:
+                    initial_value = getattr(_DEFAULT_STYLES, initial_rule_name)
+                    node_rules[initial_rule_name] = initial_value  # type: ignore[literal-required]
```

Comment:
Assigning outside of the `if: ... else: ...` simplifies the code a bit and we only get one typing ignore.

```suggestion
                if default_rules:
                    # There is a default value
                    rule_value = max(default_rules, key=get_first_item)[1]
                else:
                    rule_value = getattr(_DEFAULT_STYLES, initial_rule_name)
                node_rules[initial_rule_name] = rule_value  # type: ignore[literal-required]
```